raft/src/main/java/org/apache/kafka/raft/LeaderState.java

```diff
@@ -217,20 +220,36 @@ public boolean updateLocalState(long fetchTimestamp, LogOffsetMetadata logOffset
      * @param replicaId replica id
      * @param fetchTimestamp fetch timestamp
      * @param logOffsetMetadata new log offset and metadata
+     * @param leaderLogEndOffset current log end offset of the leader
      * @return true if the high watermark is updated too
      */
-    public boolean updateReplicaState(int replicaId,
-                                      long fetchTimestamp,
-                                      LogOffsetMetadata logOffsetMetadata) {
+    public boolean updateReplicaState(
+            int replicaId,
+            long fetchTimestamp,
+            LogOffsetMetadata logOffsetMetadata,
+            Long leaderLogEndOffset
```

Comment:
Could we use the primitive `long`? I don't think this is intended to be nullable, is it?